tests/tests.rs

```diff
@@ -0,0 +1,290 @@
+			.list_key_versions(&ListKeyVersionsRequest {
+				store_id: "store".to_string(),
+				page_size: Some(5),
+				page_token: None,
+				key_prefix: Some("k".into()),
+			})
+			.await;
+		assert!(matches!(list_result.unwrap_err(), VssError::InternalServerError { .. }));
+
+		// Verify 3 requests hit the server
+		mock_server.expect(3).assert();
+	}
+
+	#[tokio::test]
+	async fn test_internal_err_handling() {
+		let base_url = mockito::server_url();
+		let vss_client = VssClient::new(&base_url);
+
+		let error_response = ErrorResponse { error_code: 999, message: "UnknownException".to_string() };
+		let mut _mock_server = mockito::mock("POST", Matcher::Any)
+			.with_status(999)
+			.with_body(&error_response.encode_to_vec())
+			.create();
+
+		let get_request = GetObjectRequest { store_id: "store".to_string(), key: "k1".to_string() };
+		let get_result = vss_client.get_object(&get_request).await;
+		assert!(matches!(get_result.unwrap_err(), VssError::InternalError { .. }));
+
+		let put_request = PutObjectRequest {
+			store_id: "store".to_string(),
+			global_version: Some(4),
+			transaction_items: vec![KeyValue { key: "k1".to_string(), version: 2, value: b"k1v3".to_vec() }],
+		};
+		let put_result = vss_client.put_object(&put_request).await;
+		assert!(matches!(put_result.unwrap_err(), VssError::InternalError { .. }));
+
+		let list_request = ListKeyVersionsRequest {
+			store_id: "store".to_string(),
+			page_size: Some(5),
+			page_token: None,
+			key_prefix: Some("k".into()),
+		};
+		let list_result = vss_client.list_key_versions(&list_request).await;
+		assert!(matches!(list_result.unwrap_err(), VssError::InternalError { .. }));
+
+		let malformed_error_response = b"malformed";
+		_mock_server = mockito::mock("POST", Matcher::Any)
+			.with_status(409)
+			.with_body(&malformed_error_response)
+			.create();
+
+		let get_malformed_err_response = vss_client.get_object(&get_request).await;
+		assert!(matches!(get_malformed_err_response.unwrap_err(), VssError::InternalError { .. }));
+
+		let put_malformed_err_response = vss_client.put_object(&put_request).await;
+		assert!(matches!(put_malformed_err_response.unwrap_err(), VssError::InternalError { .. }));
+
+		let list_malformed_err_response = vss_client.list_key_versions(&list_request).await;
+		assert!(matches!(list_malformed_err_response.unwrap_err(), VssError::InternalError { .. }));
+
+		// Requests to endpoints are no longer mocked and will result in network error.
+		drop(_mock_server);
+
+		let get_network_err = vss_client.get_object(&get_request).await;
+		assert!(matches!(get_network_err.unwrap_err(), VssError::InternalError { .. }));
+
+		let put_network_err = vss_client.put_object(&put_request).await;
+		assert!(matches!(put_network_err.unwrap_err(), VssError::InternalError { .. }));
+
+		let list_network_err = vss_client.list_key_versions(&list_request).await;
+		assert!(matches!(list_network_err.unwrap_err(), VssError::InternalError { .. }));
```

Comment:
Wasn't clear to me why this results in an `InternalError`. In a separate commit, could you add documentation to the enum variants and configure the library as following?

```rust
#![deny(missing_docs)]
#![deny(unsafe_code)]
```